### PR TITLE
Check for nextpage to display page links for paginated posts

### PR DESCRIPTION
**Pull request by alberto-marin**
Hi,
The "core/nextpage" it's a child of "core/post-content" but within the post-content there's no reference to it.

After getting the value for `$post_id` we could see that the block `core/nextpage` exists by using something like this:

```
$blocks = parse_blocks( get_post($post_id)->post_content );
var_dump($blocks);
```
Screenshot 2021-12-30 at 20 04 01

What is missing here is the ability to detect this "child" block. After applying  the filter to `the_content` we could check and call the function that displays page links:

```
	// Check for nextpage to display page links for paginated posts
	$nextpage = has_block('nextpage') ? wp_link_pages(['echo' => 0]) : false;
	if ( false !== $nextpage ) {
		$content = get_the_content() . $nextpage;
	}
```
Screenshot 2021-12-30 at 20 12 17

I have tested it with the official wordpress themes from twentytwentynine to twentytwentytwo, and it works well.

This fix is related to this issue: https://github.com/WordPress/gutenberg/issues/29484

Note that `get_the_content()` doesn't need the `null` and `false` attributes because those are the function defaults.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by Mamaduka** [comment]
Thanks for contributing, @alberto-marin.

I'm wondering if this should be a separate block. @carolinan, @kjellr, what do you think?

**Comment by carolinan** [comment]
@Mamaduka I did not understand; the page break is already a block.

From the issue:
https://github.com/WordPress/gutenberg/issues/29484

> I have confirmed that if I add a page break block to a theme that supports full site editing, the block shows correctly in the site editor and post editor, but not the front.
> 
> The page break block also does not work with traditional/PHP based themes that does not add `wp_link_pages()`. See https://developer.wordpress.org/reference/functions/wp_link_pages/
> 
> Looks like without wp_link_pages, `<!‐‐nextpage‐‐>` which is saved in https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/nextpage/save.js#L7 is not understood?

**Comment by carolinan** [comment]
Don't we need to add this in a new .php file for nextpage?
https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/nextpage

**Comment by Mamaduka** [comment]
@carolinan, sorry for the confusion.

I meant "Page Links" as a separate/new block. The block will give theme authors more flexibility in displaying these links + styling options.

**Comment by kjellr** [comment]
> I meant "Page Links" as a separate/new block. The block will give theme authors more flexibility in displaying these links + styling options.

I think eventually, we should just follow the lead of what we do for standard query pagination. In that case, there are three separate blocks: 

- Previous Page
- Page Numbers
- Next Page

**Comment by Mamaduka** [comment]
@kjellr, this is slightly different from Query pagination blocks.

You have separate methods (`get_next_posts_link`, `paginate_links`, etc.) handling each part of the functionality for Query blocks. While for the post's pages, we have a single method that handles all the above. It is the reason I was suggesting one new block.

**Comment by kjellr** [comment]
That sounds like a technical reason, not a UX reason though — I think there's value in making pagination consistent wherever it's available so users don't have to re-learn it every time.

For the near term though, I don't have a strong preference about whether that's a separate block or not. That seems like a decent idea, but as long as this is showing up again for block themes I think we're good for now.

File: packages/block-library/src/post-content/index.php [hunk 1]
```diff
@@ -46,6 +46,11 @@ function render_block_core_post_content( $attributes, $content, $block ) {
 	// so that `the_preview` for the current post can apply.
 	// We force this behavior by omitting the third argument (post ID) from the `get_the_content`.
 	$content = get_the_content( null, false );
+	// Check for nextpage to display page links for paginated posts.
+	$nextpage = has_block( 'nextpage' ) ? wp_link_pages( array( 'echo' => 0 ) ) : false;
+	if ( false !== $nextpage ) {
+		$content = get_the_content() . $nextpage;
+	}
```
Comment (Mamaduka) [inline packages/block-library/src/post-content/index.php]:
```suggestion
	if ( has_block( 'core/nextpage' ) ) {
		$content .= wp_link_pages( array( 'echo' => 0 ) );
	}
```

@alberto-marin, a few minor suggestions.

* Let's use the prefix for block name.
* We can also append `wp_link_pages` directly, and there's no need to call `get_the_content` a second time.

Comment (Mamaduka) [reply packages/block-library/src/post-content/index.php]: It looks like there's also a small merge conflict. Once these are resolved, we can merge the PR.

Thanks for your work on this issue!

File: packages/block-library/src/post-content/index.php [hunk 2]
```diff
@@ -45,7 +45,12 @@ function render_block_core_post_content( $attributes, $content, $block ) {
 	// When inside the main loop, we want to use queried object
 	// so that `the_preview` for the current post can apply.
 	// We force this behavior by omitting the third argument (post ID) from the `get_the_content`.
-	$content = get_the_content();
+	$content = get_the_content( null, false );
```
Comment (Mamaduka) [inline packages/block-library/src/post-content/index.php]:
```suggestion
	$content = get_the_content();
```

To match the trunk.

**Review by Mamaduka — APPROVED** [review]
Changes look good. Thanks, @alberto-marin.

**Comment by Mamaduka** [comment]
@noisysocks, I'm adding a backport label for this PR. It's a small change and adds missing behavior.

**Pull request merged**